plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/TestMongoConnectorTest.java

```diff
@@ -881,6 +883,111 @@ public void testSystemSchemas()
         assertQueryReturnsEmptyResult("SHOW SCHEMAS IN mongodb LIKE 'local'");
     }
 
+    @Test(dataProvider = "nestedValuesProvider")
+    public void testFiltersOnDereferenceColumnReadsLessData(String expectedValue, String expectedType)
+    {
+        if (!isPushdownSupportedType(getQueryRunner().getTypeManager().fromSqlType(expectedType))) {
+            throw new SkipException("Type doesn't support filter pushdown");
+        }
+
+        Session sessionWithoutPushdown = Session.builder(getSession())
+                .setCatalogSessionProperty("mongodb", "projection_pushdown_enabled", "false")
+                .build();
+
+        try (TestTable table = new TestTable(
+                getQueryRunner()::execute,
+                "filter_on_projection_columns",
+                format("(col_0 ROW(col_1 %1$s, col_2 ROW(col_3 %1$s, col_4 ROW(col_5 %1$s))))", expectedType))) {
+            assertUpdate(format("INSERT INTO %1$s VALUES NULL", table.getName()), 1);
+            assertUpdate(format("INSERT INTO %1$s VALUES ROW(ROW(%2$s, ROW(%2$s, ROW(%2$s))))", table.getName(), expectedValue), 1);
+            assertUpdate(format("INSERT INTO %1$s VALUES ROW(ROW(%2$s, ROW(NULL, ROW(%2$s))))", table.getName(), expectedValue), 1);
+
+            Set<Object> expected = ImmutableSet.of(1);
+
+            assertQueryStats(
+                    getSession(),
+                    format("SELECT 1 FROM %s WHERE col_0.col_1 = %s", table.getName(), expectedValue),
```

Comment:
Can we simply use:
```
            assertThat(query("...")
                    .matches("...")
                    .isFullyPushedDown();
```                    
?

Reply:
We can not use `isFullyPushedDown` for predicate pushdown. I had a talk with @hashhar offline. Thanks @hashhar for the answer.

> isFullyPushedDown just looks at plan to verify there are no nodes between TableScan and output. It doesn't verify that the pushdown actually causes connector to read only required columns.